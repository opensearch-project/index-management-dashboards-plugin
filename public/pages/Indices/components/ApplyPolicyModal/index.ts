/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import ApplyPolicyModal from "./ApplyPolicyModal";

export default ApplyPolicyModal;
