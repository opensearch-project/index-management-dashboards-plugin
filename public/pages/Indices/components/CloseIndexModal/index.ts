/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import CloseIndexModal from "./CloseIndexModal";

export default CloseIndexModal;
