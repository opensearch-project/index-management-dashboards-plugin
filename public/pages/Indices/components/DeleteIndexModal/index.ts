/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import DeleteIndexModal from "./DeleteIndexModal";

export default DeleteIndexModal;
