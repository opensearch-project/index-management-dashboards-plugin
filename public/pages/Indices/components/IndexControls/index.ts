/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import IndexControls from "./IndexControls";

export default IndexControls;
