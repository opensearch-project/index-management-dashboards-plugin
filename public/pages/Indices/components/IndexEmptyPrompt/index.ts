/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import IndexEmptyPrompt from "./IndexEmptyPrompt";

export default IndexEmptyPrompt;
