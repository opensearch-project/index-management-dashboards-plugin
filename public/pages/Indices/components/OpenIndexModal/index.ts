/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import OpenIndexModal from "./OpenIndexModal";

export default OpenIndexModal;
