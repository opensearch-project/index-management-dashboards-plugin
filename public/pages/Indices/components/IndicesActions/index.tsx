/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import React, { useCallback, useContext, useMemo, useState } from "react";
import {
  // @ts-ignore
  Criteria,
  // @ts-ignore
  Pagination,
  EuiButton,
  EuiContextMenu,
} from "@elastic/eui";

import { ManagedCatIndex } from "../../../../../server/models/interfaces";
import ApplyPolicyModal from "../../components/ApplyPolicyModal";
import SimplePopover from "../../../../components/SimplePopover";
import { ModalConsumer } from "../../../../components/Modal";
import { CoreServicesContext } from "../../../../components/core_services";
import DeleteIndexModal from "../../components/DeleteIndexModal";
import ReindexFlyout from "../ReindexFlyout";
import { REQUEST } from "../../../../../utils/constants";
import { ReindexRequest } from "../../models/interfaces";
import IndexService from "../../../../services/IndexService";
import CommonService from "../../../../services/CommonService";
import { ServicesContext } from "../../../../services";

interface IndicesActionsProps {
  selectedItems: ManagedCatIndex[];
  onDelete?: () => void;
  indexService: IndexService;
  commonService: CommonService;
}

export default function IndicesActions(props: IndicesActionsProps) {
  const { selectedItems, onDelete, commonService } = props;
  const [deleteIndexModalVisible, setDeleteIndexModalVisible] = useState(false);
  const [isReindexFlyoutVisible, setIsReindexFlyoutVisible] = useState(false);
  const context = useContext(CoreServicesContext);
  const services = useContext(ServicesContext);

  const onDeleteIndexModalClose = () => {
    setDeleteIndexModalVisible(false);
  };

  const onDeleteIndexModalConfirm = useCallback(async () => {
    const result = await services?.commonService.apiCaller({
      endpoint: "indices.delete",
      data: {
        index: selectedItems.map((item) => item.index).join(","),
      },
    });
    if (result && result.ok) {
      onDeleteIndexModalClose();
      context?.notifications.toasts.addSuccess("Delete successfully");
      onDelete && onDelete();
    } else {
      context?.notifications.toasts.addDanger(result?.error || "");
    }
  }, [selectedItems, onDelete, services, context, onDeleteIndexModalClose]);

  const onReindexConfirmed = async (reindexRequest: ReindexRequest) => {
    let res = await commonService.apiCaller({
      endpoint: "reindex",
      method: REQUEST.POST,
      data: reindexRequest,
    });
    if (res.ok) {
      // @ts-ignore
      context?.notifications.toasts.addSuccess(`Reindex triggered successfully with taskId ${res.response.task}`);
      onCloseReindexFlyout();
    } else {
      context?.notifications.toasts.addDanger(`Reindex operation error happened ${res.error}`);
    }
  };

  const onCloseReindexFlyout = () => {
    setIsReindexFlyoutVisible(false);
  };

  const renderKey = useMemo(() => Date.now(), [selectedItems]);

  return (
    <>
      <ModalConsumer>
        {({ onShow }) => (
          <SimplePopover
            data-test-subj="More Action"
            panelPaddingSize="none"
            button={
              <EuiButton iconType="arrowDown" iconSide="right">
                Actions
              </EuiButton>
            }
          >
            <EuiContextMenu
              initialPanelId={0}
              // The EuiContextMenu has bug when testing in jest
              // the props change won't make it rerender
              key={renderKey}
              panels={[
                {
                  id: 0,
                  items: [
                    {
                      name: "Delete",
                      disabled: !selectedItems.length,
                      "data-test-subj": "Delete Action",
                      onClick: () => setDeleteIndexModalVisible(true),
                    },
                    {
                      name: "Apply policy",
                      disabled: !selectedItems.length,
                      "data-test-subj": "Apply policyButton",
                      onClick: () =>
                        onShow(ApplyPolicyModal, {
                          indices: selectedItems.map((item: ManagedCatIndex) => item.index),
                          core: context,
                        }),
                    },
                    {
                      name: "Reindex",
                      disabled: !selectedItems.length,
                      "data-test-subj": "Reindex Action",
                      onClick: () => setIsReindexFlyoutVisible(true),
                    },
                  ],
                },
              ]}
            />
          </SimplePopover>
        )}
      </ModalConsumer>
      <DeleteIndexModal
        selectedItems={selectedItems.map((item) => item.index)}
        visible={deleteIndexModalVisible}
        onClose={onDeleteIndexModalClose}
        onConfirm={onDeleteIndexModalConfirm}
      />

      {isReindexFlyoutVisible && (
        <ReindexFlyout
          {...props}
          onCloseFlyout={onCloseReindexFlyout}
          sourceIndices={selectedItems.map((item) => item.index)}
          onReindexConfirmed={onReindexConfirmed}
        />
      )}
    </>
  );
}
