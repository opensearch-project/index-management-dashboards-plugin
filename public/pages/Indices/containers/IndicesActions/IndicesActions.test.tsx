/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, waitFor } from "@testing-library/react";
// @ts-ignore
import userEvent from "@testing-library/user-event";
import { browserServicesMock, coreServicesMock } from "../../../../../test/mocks";
import IndicesActions, { IndicesActionsProps } from "./index";
import { ModalProvider } from "../../../../components/Modal";
import { ServicesContext } from "../../../../services";
import { CoreServicesContext } from "../../../../components/core_services";

function renderWithRouter(props: IndicesActionsProps) {
  return {
    ...render(
      <CoreServicesContext.Provider value={coreServicesMock}>
        <ServicesContext.Provider value={browserServicesMock}>
          <ModalProvider>
            <IndicesActions {...props} />
          </ModalProvider>
        </ServicesContext.Provider>
      </CoreServicesContext.Provider>
    ),
  };
}

describe("<IndicesActions /> spec", () => {
  it("renders the component and all the actions should be disabled when no items selected", async () => {
    const { container, getByTestId } = renderWithRouter({
      selectedItems: [],
      onDelete: () => null,
    });

    await waitFor(() => {
      expect(container.firstChild).toMatchSnapshot();
    });

    userEvent.click(document.querySelector('[data-test-subj="More Action"] button') as Element);
    await waitFor(() => {
      expect(getByTestId("Open Action")).toBeDisabled();
      expect(getByTestId("Close Action")).toBeDisabled();
      expect(getByTestId("Shrink Action")).toBeDisabled();
      expect(getByTestId("Delete Action")).toBeDisabled();
      expect(getByTestId("Apply policyButton")).toBeDisabled();
    });
  });

  it("open index by calling commonService", async () => {
    const onOpen = jest.fn();
    browserServicesMock.commonService.apiCaller = jest.fn().mockResolvedValue({ ok: true, response: {} });
    const { container, getByTestId, getByPlaceholderText } = renderWithRouter({
      selectedItems: [
        {
          "docs.count": "5",
          "docs.deleted": "2",
          health: "green",
          index: "test_index",
          pri: "1",
          "pri.store.size": "100KB",
          rep: "0",
          status: "open",
          "store.size": "100KB",
          uuid: "some_uuid",
          managed: "",
          managedPolicy: "",
          data_stream: "",
        },
      ],
      onOpen,
    });

    await waitFor(() => {
      expect(container.firstChild).toMatchSnapshot();
    });

    userEvent.click(document.querySelector('[data-test-subj="More Action"] button') as Element);
    userEvent.click(getByTestId("Open Action"));
    userEvent.click(getByTestId("Open Confirm button"));

    await waitFor(() => {
      expect(browserServicesMock.commonService.apiCaller).toHaveBeenCalledTimes(1);
      expect(browserServicesMock.commonService.apiCaller).toHaveBeenCalledWith({
        endpoint: "indices.open",
        data: {
          index: "test_index",
        },
      });
      expect(coreServicesMock.notifications.toasts.addSuccess).toHaveBeenCalledTimes(1);
      expect(coreServicesMock.notifications.toasts.addSuccess).toHaveBeenCalledWith("Open index successfully");
      expect(onOpen).toHaveBeenCalledTimes(1);
    });
  });

  it("close index by calling commonService", async () => {
    const onClose = jest.fn();
    browserServicesMock.commonService.apiCaller = jest.fn().mockResolvedValue({ ok: true, response: {} });
    const { container, getByTestId, getByPlaceholderText } = renderWithRouter({
      selectedItems: [
        {
          "docs.count": "5",
          "docs.deleted": "2",
          health: "green",
          index: "test_index",
          pri: "1",
          "pri.store.size": "100KB",
          rep: "0",
          status: "open",
          "store.size": "100KB",
          uuid: "some_uuid",
          managed: "",
          managedPolicy: "",
          data_stream: "",
        },
      ],
      onClose,
    });

    await waitFor(() => {
      expect(container.firstChild).toMatchSnapshot();
    });

    userEvent.click(document.querySelector('[data-test-subj="More Action"] button') as Element);
    userEvent.click(getByTestId("Close Action"));
    userEvent.type(getByPlaceholderText("close"), "close");
    userEvent.click(getByTestId("Close Confirm button"));

    await waitFor(() => {
      expect(browserServicesMock.commonService.apiCaller).toHaveBeenCalledTimes(1);
      expect(browserServicesMock.commonService.apiCaller).toHaveBeenCalledWith({
        endpoint: "indices.close",
        data: {
          index: "test_index",
        },
      });
      expect(coreServicesMock.notifications.toasts.addSuccess).toHaveBeenCalledTimes(1);
      expect(coreServicesMock.notifications.toasts.addSuccess).toHaveBeenCalledWith("Close index successfully");
      expect(onClose).toHaveBeenCalledTimes(1);
    });
  });

  it("shrink index by calling commonService", async () => {
    const onShrink = jest.fn();
    browserServicesMock.commonService.apiCaller = jest.fn().mockResolvedValue({ ok: true, response: {} });

    const { container, getByTestId } = renderWithRouter({
      selectedItems: [
        {
          "docs.count": "5",
          "docs.deleted": "2",
          health: "green",
          index: "test_index",
          pri: "3",
          "pri.store.size": "100KB",
          rep: "0",
          status: "open",
          "store.size": "100KB",
          uuid: "some_uuid",
          managed: "",
          managedPolicy: "",
          data_stream: "",
        },
      ],
      onShrink,
    });

    await waitFor(() => {
      expect(container.firstChild).toMatchSnapshot();
    });

    userEvent.click(document.querySelector('[data-test-subj="More Action"] button') as Element);
    userEvent.click(getByTestId("Shrink Action"));
    userEvent.type(getByTestId("targetIndexNameInput"), "test_index_shrunken");
    userEvent.click(getByTestId("shrinkIndexConfirmButton"));

    await waitFor(() => {
      expect(browserServicesMock.commonService.apiCaller).toHaveBeenCalledTimes(2);
      expect(browserServicesMock.commonService.apiCaller).toHaveBeenCalledWith({
        endpoint: "indices.shrink",
        data: {
          index: "test_index",
          target: "test_index_shrunken",
          body: {
            settings: {
              "index.number_of_shards": 1,
            },
          },
        },
      });
      expect(browserServicesMock.commonService.apiCaller).toHaveBeenCalledWith({
        endpoint: "indices.getSettings",
        data: {
          index: "test_index",
          flat_settings: true,
        },
      });
      expect(coreServicesMock.notifications.toasts.addSuccess).toHaveBeenCalledTimes(1);
      expect(coreServicesMock.notifications.toasts.addSuccess).toHaveBeenCalledWith("Shrink index successfully");
      expect(onShrink).toHaveBeenCalledTimes(1);
    });
  });

  it("delete index by calling commonService", async () => {
    const onDelete = jest.fn();
    let times = 0;
    browserServicesMock.commonService.apiCaller = jest.fn(
      async (payload): Promise<any> => {
        if (payload.endpoint === "indices.delete") {
          if (times >= 1) {
            return {
              ok: true,
              response: {},
            };
          } else {
            times++;
            return {
              ok: false,
              error: "test error",
            };
          }
        }
        return { ok: true, response: {} };
      }
    );
    const { container, getByTestId, getByPlaceholderText, findByTestId } = renderWithRouter({
      selectedItems: [
        {
          "docs.count": "5",
          "docs.deleted": "2",
          health: "green",
          index: "test_index",
          pri: "1",
          "pri.store.size": "100KB",
          rep: "0",
          status: "open",
          "store.size": "100KB",
          uuid: "some_uuid",
          managed: "",
          managedPolicy: "",
          data_stream: "",
        },
      ],
      onDelete,
    });

    await waitFor(() => {
      expect(container.firstChild).toMatchSnapshot();
    });

    userEvent.click(document.querySelector('[data-test-subj="More Action"] button') as Element);
    userEvent.click(getByTestId("Delete Action"));
    userEvent.type(getByPlaceholderText("delete"), "delete");
    userEvent.click(getByTestId("Delete Confirm button"));

    await waitFor(() => {
      expect(browserServicesMock.commonService.apiCaller).toHaveBeenCalledTimes(1);
      expect(browserServicesMock.commonService.apiCaller).toHaveBeenCalledWith({
        endpoint: "indices.delete",
        data: {
          index: "test_index",
        },
      });
      expect(coreServicesMock.notifications.toasts.addDanger).toHaveBeenCalledTimes(1);
      expect(coreServicesMock.notifications.toasts.addDanger).toHaveBeenCalledWith("test error");
      expect(onDelete).toHaveBeenCalledTimes(0);
    });

    userEvent.click(getByTestId("Delete Confirm button"));

    await waitFor(() => {
      expect(browserServicesMock.commonService.apiCaller).toHaveBeenCalledTimes(2);
      expect(coreServicesMock.notifications.toasts.addSuccess).toHaveBeenCalledTimes(1);
      expect(coreServicesMock.notifications.toasts.addSuccess).toHaveBeenCalledWith("Delete successfully");
      expect(onDelete).toHaveBeenCalledTimes(1);
    });
  });
});
