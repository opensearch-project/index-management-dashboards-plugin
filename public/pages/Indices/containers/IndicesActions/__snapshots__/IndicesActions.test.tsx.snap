// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IndicesActions /> spec Split index by calling commonService 1`] = `
<div
  class="euiPopover euiPopover--anchorDownCenter"
  data-test-subj="moreAction"
>
  <div
    class="euiPopover__anchor"
  >
    <button
      class="euiButton euiButton--primary euiButton--small"
      type="button"
    >
      <span
        class="euiButtonContent euiButtonContent--iconRight euiButton__content"
      >
        EuiIconMock
        <span
          class="euiButton__text"
        >
          Actions
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`<IndicesActions /> spec clear cache for all indexes failed if some indexes are blocked 1`] = `
<div
  class="euiPopover euiPopover--anchorDownCenter"
  data-test-subj="moreAction"
>
  <div
    class="euiPopover__anchor"
  >
    <button
      class="euiButton euiButton--primary euiButton--small"
      type="button"
    >
      <span
        class="euiButtonContent euiButtonContent--iconRight euiButton__content"
      >
        EuiIconMock
        <span
          class="euiButton__text"
        >
          Actions
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`<IndicesActions /> spec clear cache for all indexes successfully by calling commonService 1`] = `
<div
  class="euiPopover euiPopover--anchorDownCenter"
  data-test-subj="moreAction"
>
  <div
    class="euiPopover__anchor"
  >
    <button
      class="euiButton euiButton--primary euiButton--small"
      type="button"
    >
      <span
        class="euiButtonContent euiButtonContent--iconRight euiButton__content"
      >
        EuiIconMock
        <span
          class="euiButton__text"
        >
          Actions
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`<IndicesActions /> spec clear cache for mulitple indexes by calling commonService 1`] = `
<div
  class="euiPopover euiPopover--anchorDownCenter"
  data-test-subj="moreAction"
>
  <div
    class="euiPopover__anchor"
  >
    <button
      class="euiButton euiButton--primary euiButton--small"
      type="button"
    >
      <span
        class="euiButtonContent euiButtonContent--iconRight euiButton__content"
      >
        EuiIconMock
        <span
          class="euiButton__text"
        >
          Actions
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`<IndicesActions /> spec close index by calling commonService 1`] = `
<div
  class="euiPopover euiPopover--anchorDownCenter"
  data-test-subj="moreAction"
>
  <div
    class="euiPopover__anchor"
  >
    <button
      class="euiButton euiButton--primary euiButton--small"
      type="button"
    >
      <span
        class="euiButtonContent euiButtonContent--iconRight euiButton__content"
      >
        EuiIconMock
        <span
          class="euiButton__text"
        >
          Actions
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`<IndicesActions /> spec delete index by calling commonService 1`] = `
<div
  class="euiPopover euiPopover--anchorDownCenter"
  data-test-subj="moreAction"
>
  <div
    class="euiPopover__anchor"
  >
    <button
      class="euiButton euiButton--primary euiButton--small"
      type="button"
    >
      <span
        class="euiButtonContent euiButtonContent--iconRight euiButton__content"
      >
        EuiIconMock
        <span
          class="euiButton__text"
        >
          Actions
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`<IndicesActions /> spec filter indices failed when clearing caches for multiple indices 1`] = `
<div
  class="euiPopover euiPopover--anchorDownCenter"
  data-test-subj="moreAction"
>
  <div
    class="euiPopover__anchor"
  >
    <button
      class="euiButton euiButton--primary euiButton--small"
      type="button"
    >
      <span
        class="euiButtonContent euiButtonContent--iconRight euiButton__content"
      >
        EuiIconMock
        <span
          class="euiButton__text"
        >
          Actions
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`<IndicesActions /> spec filter some closed or blocked indices when clearing caches for multiple indices 1`] = `
<div
  class="euiPopover euiPopover--anchorDownCenter"
  data-test-subj="moreAction"
>
  <div
    class="euiPopover__anchor"
  >
    <button
      class="euiButton euiButton--primary euiButton--small"
      type="button"
    >
      <span
        class="euiButtonContent euiButtonContent--iconRight euiButton__content"
      >
        EuiIconMock
        <span
          class="euiButton__text"
        >
          Actions
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`<IndicesActions /> spec open index by calling commonService 1`] = `
<div
  class="euiPopover euiPopover--anchorDownCenter"
  data-test-subj="moreAction"
>
  <div
    class="euiPopover__anchor"
  >
    <button
      class="euiButton euiButton--primary euiButton--small"
      type="button"
    >
      <span
        class="euiButtonContent euiButtonContent--iconRight euiButton__content"
      >
        EuiIconMock
        <span
          class="euiButton__text"
        >
          Actions
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`<IndicesActions /> spec refresh selected indexes by calling commonService 1`] = `
<body
  class="euiBody-hasOverlayMask"
>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <div
      class="euiPopover euiPopover--anchorDownCenter"
      data-test-subj="moreAction"
    >
      <div
        class="euiPopover__anchor"
      >
        <button
          class="euiButton euiButton--primary euiButton--small"
          type="button"
        >
          <span
            class="euiButtonContent euiButtonContent--iconRight euiButton__content"
          >
            EuiIconMock
            <span
              class="euiButton__text"
            >
              Actions
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    class="euiOverlayMask euiOverlayMask--aboveHeader"
  >
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <div
        class="euiModal euiModal--maxWidth-default"
        tabindex="0"
      >
        <button
          aria-label="Closes this modal window"
          class="euiButtonIcon euiButtonIcon--text euiButtonIcon--empty euiButtonIcon--xSmall euiModal__closeIcon"
          type="button"
        >
          EuiIconMock
        </button>
        <div
          class="euiModal__flex"
        >
          <div
            class="euiModalHeader"
          >
            <div
              class="euiModalHeader__title"
            >
              <div
                class="euiText euiText--small"
              >
                <h2>
                  Refresh 
                  indexes
                </h2>
              </div>
            </div>
          </div>
          <div
            class="euiModalBody"
          >
            <div
              class="euiModalBody__overflow"
            >
              <div
                style="line-height: 1.5;"
              >
                <div
                  class="euiText euiText--small"
                >
                  <p>
                    The following index
                     will be refreshed.
                  </p>
                </div>
                <ul
                  style="list-style-type: disc; list-style-position: inside;"
                >
                  <div
                    class="euiText euiText--small"
                  >
                    <li
                      data-test-subj="UnblockedItem-unblocked_index"
                    >
                      unblocked_index
                    </li>
                  </div>
                </ul>
                <div
                  class="euiSpacer euiSpacer--l"
                />
                <div
                  class="euiCallOut euiCallOut--warning euiCallOut--small"
                >
                  <div
                    class="euiCallOutHeader"
                  >
                    EuiIconMock
                    <span
                      class="euiCallOutHeader__title"
                    >
                      The following index cannot be refreshed because they are either closed or in red status.
                    </span>
                  </div>
                  <div
                    class="euiText euiText--extraSmall"
                  >
                    <div
                      class="euiTextColor euiTextColor--default"
                    >
                      <p />
                      <ul
                        style="list-style-type: disc; list-style-position: inside;"
                      >
                        <li
                          data-test-subj="BlockedItem-blocked_index"
                        >
                          blocked_index
                        </li>
                      </ul>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="euiModalFooter"
          >
            <button
              class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--small"
              type="button"
            >
              <span
                class="euiButtonContent euiButtonEmpty__content"
              >
                <span
                  class="euiButtonEmpty__text"
                >
                  Cancel
                </span>
              </span>
            </button>
            <button
              class="euiButton euiButton--primary euiButton--small euiButton--fill"
              data-test-subj="refreshConfirmButton"
              type="button"
            >
              <span
                class="euiButtonContent euiButton__content"
              >
                <span
                  class="euiButton__text"
                >
                  Refresh
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`<IndicesActions /> spec renders flush component 1`] = `
HTMLCollection [
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <div
      class="euiPopover euiPopover--anchorDownCenter"
      data-test-subj="moreAction"
    >
      <div
        class="euiPopover__anchor"
      >
        <button
          class="euiButton euiButton--primary euiButton--small"
          type="button"
        >
          <span
            class="euiButtonContent euiButtonContent--iconRight euiButton__content"
          >
            EuiIconMock
            <span
              class="euiButton__text"
            >
              Actions
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>,
  <div
    class="euiOverlayMask euiOverlayMask--aboveHeader"
  >
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <div
        class="euiModal euiModal--maxWidth-default"
        tabindex="0"
      >
        <button
          aria-label="Closes this modal window"
          class="euiButtonIcon euiButtonIcon--text euiButtonIcon--empty euiButtonIcon--xSmall euiModal__closeIcon"
          type="button"
        >
          EuiIconMock
        </button>
        <div
          class="euiModal__flex"
        >
          <div
            class="euiModalHeader"
          >
            <div
              class="euiModalHeader__title"
              data-test-subj="flushModalTitle"
            >
              <div
                class="euiText euiText--small"
              >
                <h2>
                  Flush 
                  indexes
                </h2>
              </div>
            </div>
          </div>
          <div
            class="euiModalBody"
          >
            <div
              class="euiModalBody__overflow"
            >
              <div
                style="line-height: 1.5;"
              >
                <div
                  class="euiText euiText--small"
                >
                  <p>
                    The following indexes will be flushed:
                  </p>
                </div>
                <ul
                  style="list-style-type: disc; list-style-position: inside;"
                >
                  <div
                    class="euiText euiText--small"
                  >
                    <li>
                      test_index2
                    </li>
                  </div>
                  <div
                    class="euiText euiText--small"
                  >
                    <li>
                      test_index3
                    </li>
                  </div>
                </ul>
                <div
                  class="euiSpacer euiSpacer--l"
                />
                <div
                  class="euiCallOut euiCallOut--warning euiCallOut--small"
                  data-test-subj="flushBlockedCallout"
                >
                  <div
                    class="euiText euiText--extraSmall"
                  >
                    <div
                      class="euiTextColor euiTextColor--default"
                    >
                      <div
                        class="euiText euiText--small"
                      >
                        <p>
                          The following indexes will not be flushed because they are either closed or in red status:
                        </p>
                      </div>
                      <ul
                        style="list-style-type: disc; list-style-position: inside;"
                      >
                        <div
                          class="euiText euiText--small"
                        >
                          <li>
                            test_index1
                          </li>
                        </div>
                        <div
                          class="euiText euiText--small"
                        >
                          <li>
                            test_index4
                          </li>
                        </div>
                      </ul>
                    </div>
                  </div>
                </div>
                <div
                  class="euiSpacer euiSpacer--l"
                />
              </div>
            </div>
          </div>
          <div
            class="euiModalFooter"
          >
            <button
              class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--small"
              data-test-subj="flushCancelButton"
              type="button"
            >
              <span
                class="euiButtonContent euiButtonEmpty__content"
              >
                <span
                  class="euiButtonEmpty__text"
                >
                  Cancel
                </span>
              </span>
            </button>
            <button
              class="euiButton euiButton--primary euiButton--small euiButton--fill"
              data-test-subj="flushConfirmButton"
              type="button"
            >
              <span
                class="euiButtonContent euiButton__content"
              >
                <span
                  class="euiButton__text"
                >
                  Flush
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>,
]
`;

exports[`<IndicesActions /> spec renders the component and all the actions should be disabled when no items selected 1`] = `
<div
  class="euiPopover euiPopover--anchorDownCenter"
  data-test-subj="moreAction"
>
  <div
    class="euiPopover__anchor"
  >
    <button
      class="euiButton euiButton--primary euiButton--small"
      type="button"
    >
      <span
        class="euiButtonContent euiButtonContent--iconRight euiButton__content"
      >
        EuiIconMock
        <span
          class="euiButton__text"
        >
          Actions
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`<IndicesActions /> spec shrink action is disabled if multiple indices are selected 1`] = `
<div
  class="euiPopover euiPopover--anchorDownCenter"
  data-test-subj="moreAction"
>
  <div
    class="euiPopover__anchor"
  >
    <button
      class="euiButton euiButton--primary euiButton--small"
      type="button"
    >
      <span
        class="euiButtonContent euiButtonContent--iconRight euiButton__content"
      >
        EuiIconMock
        <span
          class="euiButton__text"
        >
          Actions
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`<IndicesActions /> spec shrink action is disabled if the selected index is data_stream 1`] = `
<div
  class="euiPopover euiPopover--anchorDownCenter"
  data-test-subj="moreAction"
>
  <div
    class="euiPopover__anchor"
  >
    <button
      class="euiButton euiButton--primary euiButton--small"
      type="button"
    >
      <span
        class="euiButtonContent euiButtonContent--iconRight euiButton__content"
      >
        EuiIconMock
        <span
          class="euiButton__text"
        >
          Actions
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`<IndicesActions /> spec shrink index by calling commonService 1`] = `
<div
  class="euiPopover euiPopover--anchorDownCenter"
  data-test-subj="moreAction"
>
  <div
    class="euiPopover__anchor"
  >
    <button
      class="euiButton euiButton--primary euiButton--small"
      type="button"
    >
      <span
        class="euiButtonContent euiButtonContent--iconRight euiButton__content"
      >
        EuiIconMock
        <span
          class="euiButton__text"
        >
          Actions
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`<IndicesActions /> spec split action is disabled if multiple indices are selected 1`] = `
<div
  class="euiPopover euiPopover--anchorDownCenter"
  data-test-subj="moreAction"
>
  <div
    class="euiPopover__anchor"
  >
    <button
      class="euiButton euiButton--primary euiButton--small"
      type="button"
    >
      <span
        class="euiButtonContent euiButtonContent--iconRight euiButton__content"
      >
        EuiIconMock
        <span
          class="euiButton__text"
        >
          Actions
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`<IndicesActions /> spec split action is disabled if the selected index is data_stream 1`] = `
<div
  class="euiPopover euiPopover--anchorDownCenter"
  data-test-subj="moreAction"
>
  <div
    class="euiPopover__anchor"
  >
    <button
      class="euiButton euiButton--primary euiButton--small"
      type="button"
    >
      <span
        class="euiButtonContent euiButtonContent--iconRight euiButton__content"
      >
        EuiIconMock
        <span
          class="euiButton__text"
        >
          Actions
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`<IndicesActions /> spec unable to clear cache when clearing cache for a closed or blocked index 1`] = `
<div
  class="euiPopover euiPopover--anchorDownCenter"
  data-test-subj="moreAction"
>
  <div
    class="euiPopover__anchor"
  >
    <button
      class="euiButton euiButton--primary euiButton--small"
      type="button"
    >
      <span
        class="euiButtonContent euiButtonContent--iconRight euiButton__content"
      >
        EuiIconMock
        <span
          class="euiButton__text"
        >
          Actions
        </span>
      </span>
    </button>
  </div>
</div>
`;
